tieredhashing/tiered_hashing.go

```diff
@@ -428,6 +429,18 @@ func (t *TieredHashing) UpdateMainTierWithTopN() (mainToUnknown, unknownToMain i
 		}
 	}
 
+	// if we don't have enough nodes for the main pool, compromise on number of observations required and fill it up
+	if t.mainSet.Size() < t.cfg.MaxMainTierSize {
```

Comment:
shouldn't this be a `MinMainTierSize` threshold. otherwise it's improperly named and we're essentially forcing an exact singular `MainTierSize`?

Reply:
@willscott A subsequent PR will be to make this pool unbounded as we've discussed so all performant nodes can join it. Okay if we deal with the naming then ?